ironfish/src/fileStores/config.ts

```diff
@@ -338,9 +349,13 @@ export class Config extends KeyStore<ConfigOptions> {
     return this.files.join(this.storage.dataDir, 'temp')
   }
 
+  isBootstrapNodesSet(): boolean {
+    return this.keysLoaded.has('bootstrapNodes') || this.overrides.bootstrapNodes?.length !== 0
```

Comment:
This could use a test. The KeyStore does some prototypal inheritance with defaults+loaded+overrides, so you'll want to double-check that you're only checking the `overrides` object (and not its parents) for whether bootstrapNodes is set.

Reply:
Added tests for this function

Reply:
Sorry, I didn't see this before, but this should check if `bootstrapNodes` is set in `overrides`, not just whether it has values of 0, so that we can allow people to pass an empty array. You can test this by changing the default config to `bootstrapNodes: ['']`, then making sure the `config.test.ts` tests still pass.

You can check if just the object (and not its prototype chain) has a property. It's just a bit inconvenient in JS: `Object.prototype.hasOwnProperty.call(this.overrides, 'bootstrapNodes')`

Reply:
Updated to check this, I kept the default as `[]` but changed the test to test the case when people pass an empty array.